src/main.ts

```diff
@@ -1,152 +1,55 @@
-        input: Buffer.alloc(masterPassword.length, masterPassword)
-      }
+  core.startGroup('Signing in to 1Password')
+  try {
+    await onePassword.signIn(
+      signInAddress,
+      emailAddress,
+      secretKey,
+      masterPassword
     )
-    const session = output.toString().trim()
-    core.setSecret(session)
-
-    env.OP_SESSION_github_action = session
-
-    const itemRequests = parsing.parseItemRequestsInput(itemRequestsString)
-
-    for (const itemRequest of itemRequests) {
-      const itemsJSON = await execWithOutput(
-        'op',
-        ['list', 'items', '--vault', itemRequest.vault],
-        {env}
-      )
-
-      const items: Item[] = JSON.parse(itemsJSON)
-      const uuid = items
-        .filter(item => item.overview.title === itemRequest.name)
-        .map(item => item.uuid)[0]
-
-      const itemJSON = await execWithOutput(
-        'op',
-        ['get', 'item', uuid, '--vault', itemRequest.vault],
-        {env}
-      )
-      const item: Item = JSON.parse(itemJSON)
-
-      switch (item.templateUuid) {
-        // Item
-        case '001': {
-          const username = (item.details.fields ?? []).filter(
-            field => field.designation === 'username'
-          )[0].value
-          const password = (item.details.fields ?? []).filter(
-            field => field.designation === 'password'
-          )[0].value
-
-          const usernameOutputName = `${itemRequest.outputName}_username`
-          core.setOutput(usernameOutputName, username)
-          const passwordOutputName = `${itemRequest.outputName}_password`
-          core.setSecret(password)
-          core.setOutput(passwordOutputName, password)
-
-          break
-        }
-        // Password
-        case '005': {
-          const password = item.details.password
-          if (password === undefined) {
-            throw new Error(
-              'Expected string for property item.details.password, got undefined.'
-            )
-          }
-
-          const passwordOutputName = `${itemRequest.outputName}_password`
-          core.setSecret(password)
-          core.setOutput(passwordOutputName, password)
-
-          break
-        }
-        // Document
-        case '006': {
-          const filename = item.details.documentAttributes?.fileName
-          if (filename === undefined) {
-            throw new Error(
-              'Expected string for property document.details.documentAttributes?.filename, got undefined.'
-            )
-          }
-
-          await exec.exec(
-            'op',
-            ['get', 'document', uuid, '--output', filename],
-            {env}
-          )
-
-          const documentOutputName = `${itemRequest.outputName}_filename`
-          core.setOutput(documentOutputName, filename)
-
-          break
-        }
-      }
-    }
+  } catch (error) {
+    core.setFailed(`Error signing in to 1Password: ${error.message}`)
+    return
+  }
+  core.endGroup()
 
-    core.info('Signing out of op')
```

Comment:
Looks like we might have lost the change from #43 when rebasing?

Reply:
Added back in.